src/web/src/pages/CoursePage.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div v-if="ready" class="container w-100 mb-4">
+    <b-row>
+      <b-col>
+        <h1 class="mt-4">{{ courseTitle }}</h1>
+        <h4 class="mb-1">{{ courseName }}</h4>
+      </b-col>
+    </b-row>
+    <b-row>
+      <b-col>
+        {{
+          generateRequirementsText(
+            courseObj.prerequisites,
+            courseObj.corequisites,
+            courseObj.raw_precoreqs
+          )
+        }}
+      </b-col>
+    </b-row>
+    <b-row>
+      <b-col class="mb-4">
+        <br />
+        {{ courseObj.description }}
+      </b-col>
+    </b-row>
+    <b-button :to="backRoute">Back</b-button>
+  </div>
+  <div v-else>
+    <b-spinner></b-spinner>
+    <strong class="m-2">Loading Course...</strong>
+  </div>
+</template>
+
+<script>
+import { getCourses } from "../services/YacsService";
+import { getDefaultSemester } from "@/services/AdminService";
+import { generateRequirementsText } from "@/utils";
+
+export default {
+  name: "CoursePage",
+  data() {
+    return {
+      ready: false,
+      courseName: this.$route.params.course,
+      courseTitle: String,
+      courseObj: {},
+      allCoursePreReqs: {},
+      selectedSemester: String,
+      backRoute: String,
+    };
+  },
+  methods: {
+    generateRequirementsText,
+  },
+  async created() {
+    const querySemester = this.$route.query.semester;
+    this.selectedSemester =
+      querySemester && querySemester != "null"
+        ? querySemester
+        : await getDefaultSemester();
+    getCourses(this.selectedSemester).then((courses) => {
+      for (const c of courses) {
+        this.allCoursePreReqs[c.name] = c.prerequisites;
+        if (c.name === this.courseName) {
+          this.courseTitle = c.title;
+          this.courseObj = c;
+        }
+      }
+      this.backRoute = "/explore/" + this.courseObj.department;
+      this.ready = true;
+    });
+  },
+  computed: {
+    prereqTree() {
+      let treeData = {};
+      treeData.name = this.courseName;
+      return treeData;
+    },
+  },
+};
+</script>
+
+<style>
+.container {
```

Comment:
if this is the only style for this, would prefer adding the class `text-center` instead

Reply:
That class doesn't seem to work for some reason? Now that I'm looking at it left justified again I think I prefer it anyway. I'll delete the style in favor of having it left justified instead of centered.